Libraries/MLXLMCommon/Documentation.docc/porting.md

```diff
@@ -0,0 +1,700 @@
+# Porting Models
+
+How to make new models using `mlx-swift`.
+
+There are a number of ways to implement new models in `MLX` (Swift):
+
+- built from scratch
+- ported from other ML frameworks
+    - [MLX Documentation](https://ml-explore.github.io/mlx/build/html/examples/llama-inference.html)
+- ported from Python, [e.g. `mlx-lm`](https://github.com/ml-explore/mlx-lm/tree/main/mlx_lm/models)
+
+This document talks primarily about the latter.
+
+## Porting Models from MLX (Python)
+
+Let's consider a concrete example,
+[Gemma.py](https://github.com/ml-explore/mlx-lm/blob/main/mlx_lm/models/gemma.py).  For
+reference, here is the current port
+[Gemma.swift](https://github.com/ml-explore/mlx-swift-examples/blob/main/Libraries/MLXLLM/Models/Gemma.swift).
+
+### Imports
+
+When creating the new model you need to import the right modules -- typically these are sufficient:
+
+```swift
+import Foundation
+import MLX
+import MLXFast
+import MLXLMCommon
+import MLXNN
+```
+
+- Foundation
+    - used for standard Swift features like Codable -- this let's us easily read JSON configuration
+- MLX
+    - the base [MLXArray](https://swiftpackageindex.com/ml-explore/mlx-swift/main/documentation/mlx) framework
+- MLXFast
+    - specialized and optimized functions for e.g. `rmsNorm`
+    - Note: this becomes part of `MLX` in current releases
+- MLXLMCommon
+    - language model support code (this library)
+    - provides weight loading, token generation, etc.
+- MLXNN
+    - the base [Module and NN](https://swiftpackageindex.com/ml-explore/mlx-swift/main/documentation/mlxnn) package for MLX
+
+### Configuration
+
+Next I typically port the configuration type from python -- you can see it at the top of the `gemma.py` file:
```

Comment:
Nit: the first-person here jumped out as not matching the rest of the document. Maybe change to:
```suggestion
Next port the configuration type from python -- you can see it at the top of the `gemma.py` file:
```